hazelcast/serialization/portable/classdef.py

```diff
@@ -210,71 +382,219 @@ def add_portable_array_field(self, field_name, class_def):
+
+        Returns:
+            ClassDefinitionBuilder: Itself for chaining.
+
+        Raises:
+            HazelcastSerializationError: If this method is called
+                after :func:`build` or a field with the same
+                name is already registered.
+        """
         self._add_field_by_type(field_name, FieldType.LONG_ARRAY, self.version)
         return self
 
     def add_float_array_field(self, field_name):
+        """Adds the field with the float array type to the
+        class definition.
+
+        Args:
+            field_name (str): Name of the field to add.
+
+        Returns:
+            ClassDefinitionBuilder: Itself for chaining.
+
+        Raises:
+            HazelcastSerializationError: If this method is called
+                after :func:`build` or a field with the same
+                name is already registered.
+        """
         self._add_field_by_type(field_name, FieldType.FLOAT_ARRAY, self.version)
         return self
 
     def add_double_array_field(self, field_name):
+        """Adds the field with the double array type to the
+        class definition.
+
+        Args:
+            field_name (str): Name of the field to add.
+
+        Returns:
+            ClassDefinitionBuilder: Itself for chaining.
+
+        Raises:
+            HazelcastSerializationError: If this method is called
+                after :func:`build` or a field with the same
+                name is already registered.
+        """
         self._add_field_by_type(field_name, FieldType.DOUBLE_ARRAY, self.version)
         return self
 
     def add_string_array_field(self, field_name):
+        """Adds the field with the string array type to the
+        class definition.
+
+        Args:
+            field_name (str): Name of the field to add.
+
+        Returns:
+            ClassDefinitionBuilder: Itself for chaining.
+
+        Raises:
+            HazelcastSerializationError: If this method is called
+                after :func:`build` or a field with the same
+                name is already registered.
+        """
         self._add_field_by_type(field_name, FieldType.STRING_ARRAY, self.version)
         return self
 
     def add_utf_array_field(self, field_name):
-        """
+        """Adds the field with the string array type to the
+        class definition.
+
+        Args:
+            field_name (str): Name of the field to add.
+
+        Returns:
+            ClassDefinitionBuilder: Itself for chaining.
+
+        Raises:
+            HazelcastSerializationError: If this method is called
+                after :func:`build` or a field with the same
+                name is already registered.
+
         .. deprecated:: 4.1
             This method is deprecated and will be removed in the
             next major version. Use :func:`add_string_array_field` instead.
         """
         return self.add_string_array_field(field_name)
 
     def add_field_def(self, field_def):
```

Comment:
I think this is public by accident as it is in java. Can we add documentation to tell the users that this is private(I am assuming we can not delete it anymore)

Can we add a documentation about the order that fields added should be consistent across all languages and also it should be consistent with write order in `writePortable` on ClassDefinitionBuilder ?

Reply:
Added a warning to this method and a note about the field order in the ClassDefinitionBuilder docstring